src/repository/models.py

```diff
@@ -0,0 +1,1033 @@
+            kwargs={'preprint_id': self.id,}
+        )
+
+        return url
+
+
+class KeywordPreprint(models.Model):
+    keyword = models.ForeignKey("submission.Keyword")
+    preprint = models.ForeignKey(Preprint)
+    order = models.PositiveIntegerField(default=1)
+
+    class Meta:
+        ordering = ["order"]
+        unique_together = ('keyword', 'preprint')
+
+    def __str__(self):
+        return self.keyword.word
+
+    def __repr__(self):
+        return "KeywordPreprint(%s, %d)" % (self.keyword.word, self.preprint.id)
+
+
+class PreprintFile(models.Model):
+    preprint = models.ForeignKey(Preprint)
+    file = models.FileField(
+        upload_to=preprint_file_upload,
+        storage=preprint_file_store,
+    )
+    original_filename = models.TextField()
+    uploaded = models.DateTimeField(default=timezone.now)
+    mime_type = models.CharField(
+        max_length=255,
+        blank=True,
+        null=True,
+    )
+    size = models.PositiveIntegerField(default=0)
+
+    def filename(self):
+        return os.path.basename(self.file.name)
+
+    @property
+    def uuid_filename(self):
+        return self.filename()
+
+    def get_file_mime_type(self):
+        return files.file_path_mime(self.file.path)
+
+    def path_parts(self):
+        path = os.path.dirname(os.path.abspath(self.file.path))
+        return path
+
+    def reverse_kwargs(self):
+        return {
+            'preprint_id': self.preprint.pk,
+            'file_id': self.pk,
+        }
+
+    def download_url(self):
+        return reverse(
+            'repository_download_file',
+            kwargs=self.reverse_kwargs(),
+        )
+
+    def contents(self):
+        file = open(self.file.path, mode='r')
+        contents = file.read()
+        file.close()
+        return contents
+
+
+class PreprintSupplementaryFile(models.Model):
+    preprint = models.ForeignKey(Preprint)
+    url = models.URLField()
+    label = models.CharField(max_length=200, verbose_name=_('Label'), default='Supplementary File')
+    order = models.PositiveIntegerField(default=0)
+
+    class Meta:
+        ordering = ('order',)
+        unique_together = ('url', 'preprint')
+
+
+class PreprintAccess(models.Model):
+    preprint = models.ForeignKey(Preprint)
+    file = models.ForeignKey(PreprintFile, blank=True, null=True)
+    identifier = models.TextField(blank=True, null=True)
+    accessed = models.DateTimeField(auto_now_add=True)
+    country = models.ForeignKey('core.Country', blank=True, null=True)
+
+    @property
+    def access_type(self):
+        if self.file:
+            return 'download'
+        return 'view'
+
+
+class PreprintAuthorManager(models.Manager):
+    def get_queryset(self):
+        return super().get_queryset().select_related('author')
+
+
+class PreprintAuthor(models.Model):
+    preprint = models.ForeignKey('Preprint')
+    account = models.ForeignKey('core.Account', null=True)
+    author = models.ForeignKey('Author', blank=True, null=True)
```

Comment:
can we document somehow the difference between these two? I'm assuming they behave similar to frozen authors in journals, but the implementation is different in this case (e.g There are 3 affiliation records for an "author": one in account, one in Author and one in PreprintAuthor)